### PR TITLE
Add some information on cross compilation

### PR DESCRIPTION
**Pull request by brianmcgillion**
Signed-off-by: Brian McGillion <brian.mcgillion@tii.ae>

### CODE REVIEW

**Review by vilvo — APPROVED** [review]
lgtm

File: docs/src/build_config/cross_compilation.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Cross Compilation
```
Comment (jenninikko) [inline docs/src/build_config/cross_compilation.md:1]: Is this new section is going to be under the Build Configurations chapter?
Like this:
-- Build Configurations
-------Cross Compilation
If yes,  you need to update the SUMMARY.file as well. Some general instructions are here: https://github.com/tiiuae/ghaf/blob/main/docs/README.md#working-with-files.

File: README.md [hunk 2]
```diff
@@ -37,6 +37,19 @@ then flash the resulting image eg. to a SD card or USB drive
     dd if=./result/nixos.img of=/dev/<YOUR_USB_DRIVE> bs=32M
 
 
+### Cross compilation complications
+
+As there are some packages that are not cross-compilation aware you may set the following in you configuration.nix to enable binfmt.
```
Comment (jenninikko) [inline README.md:42]: As there are some packages that are not cross-compilation aware **,** you **can** set the following in you `configuration.nix` to enable binfmt.

File: docs/src/build_config/cross_compilation.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+Ghaf is targeted at a range of devices and form factors that support different Instruction Set Architectures (ISA). Many small form-factor edge devices are not powerful enough to compile the needed applications or operating systems that run on them. As the most common ISA used in desktops and servers is ``x\_86`` this will generally require that the code is cross-compiled for target ISA e.g. ``AArch64`` or ``RISC-V``.
```
Comment (jenninikko) [inline docs/src/build_config/cross_compilation.md:3]: ...is ``x\_86`` **,** this...

**Review by jenninikko — CHANGES_REQUESTED** [review]
Hei Brian, this is a good one. Could not skip some commas and hyphens, and SUMMARY.md needs to be updated. I also keep forgetting about it all the time :)

File: docs/src/build_config/cross_compilation.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+or
+
+    boot.kernelModules = [ "kvm-intel" ];
+    
+depending on whether your development host is running ``AMD`` or ``Intel`` version of ``x\_86``.
+
+## Future cross compilation support
+
+This will involve working with upstream package maintainers to ensure that the packages are cross compilation aware. This will be addressed on a package by package basis.
```
[the remark this answers is not in the document]
Comment (brianmcgillion) [reply docs/src/build_config/cross_compilation.md:30]: I only speak the language. what is an adjective let alone compound adjective :)

**Review by jenninikko — APPROVED** [review]

**Pull request merged**